### PR TITLE
Fix builds with rocBLAS that does not support F8

### PR DESCRIPTION
**Pull request by atamazov**
By-product of triaging the issue with transposed convolutions.

----
[Attribution] @junliume @JehandadKhan
- https://github.com/ROCmSoftwarePlatform/MIOpen/labels/quality
- https://github.com/ROCmSoftwarePlatform/MIOpen/labels/urgency_high
- Proposed reviewers: @JehandadKhan

### CODE REVIEW

**Comment by junliume** [comment]
@atamazov Is your FIN branch left behind? The CI is failing and maybe you just need to update it from most recent develop branch:
```
/var/jenkins/workspace/MLLibs_MIOpen_PR-2480/fin/src/include/conv_fin.hpp:116:57: error: use of undeclared identifier 'miopenInt8x4'; did you mean 'miopenInt8'?

        return (data_type == miopenInt8 || data_type == miopenInt8x4) ? miopenFloat : data_type;

                                                        ^~~~~~~~~~~~

                                                        miopenInt8
```

**Comment by atamazov** [comment]
@junliume Thanks! I need to revert fin changes here (accidentally sneaked).

**Comment by atamazov** [comment]
@junliume @JehandadKhan Accidental changes of fin reverted. PR updated from the recent develop. Ready for the next review & CI testing round.

File: src/gemm_v2.cpp [hunk 1]
```diff
@@ -70,6 +70,8 @@
 /// Let's pass literal value as a workaround; there should be no harm.
 #define USE_GEMM_FLAGS_FP16_ALT_IMPL_242 (MIOPEN_ROCBLAS_VERSION_FLAT == 2042000)
 
+#define F8_SUPPORTED (MIOPEN_ROCBLAS_VERSION_FLAT > 3000000)
```
Comment (JehandadKhan) [inline src/gemm_v2.cpp:73]: @apwojcik Please note, I recall we were discussing this.

Comment (JehandadKhan) [inline src/gemm_v2.cpp]: @atamazov  Why not reuse the existing macro `USE_ROCBLAS_GEMM_EX3` ?

**Review by JehandadKhan — CHANGES_REQUESTED** [review]

File: src/gemm_v2.cpp (hunk 1 above)
Comment (atamazov) [reply src/gemm_v2.cpp]: @JehandadKhan I've missed that is has the same semantics. Let' me generalize the code.

Comment (atamazov) [reply src/gemm_v2.cpp]: @JehandadKhan [resolved]

**Comment by atamazov** [comment]
@junliume @JehandadKhan Ready for the next review & CI testing round.

**Review by JehandadKhan — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Pull request merged**